### PR TITLE
[CF-550] XSLT changes to support archive node

### PR DESCRIPTION
**Pull request by ChandraAddala**

### CODE REVIEW

**Comment by ChandraAddala** [comment]
@shintasmith @usnavi Pls review.

File: src/main/xsl/xml2json-feeds.xsl [hunk 1]
```diff
@@ -151,6 +151,20 @@
...
+                <xsl:text>"</xsl:text><xsl:value-of select="local-name()"/><xsl:text>" : {</xsl:text>
+                <xsl:text>"</xsl:text>@type" : "<xsl:value-of select="namespace-uri()"/><xsl:text>"</xsl:text>
+                <xsl:text>}</xsl:text>
+
+                <xsl:variable name="sibs"
+                              select="following-sibling::*[not(self::atom:category) and
+                    not(self::atom:entry) and
+                    not(self::atom:feed) and
+                    not(self::atom:link)]"/>
+                <xsl:if test="count($sibs) &gt; 0">, </xsl:if>
```
Comment (shintasmith) [inline src/main/xsl/xml2json-feeds.xsl]: Should we have different unit tests that cover this logic? I.e: put the ```fn:archive``` node at the beginning (first child element of ```feed```), in the middle, and at the end?

File: src/main/xsl/xml2json-feeds.xsl [hunk 2]
```diff
@@ -151,6 +151,20 @@
...
+            <!-- handles the archive element -->
+            <xsl:when test="cldfeeds:isArchiveNode( local-name(), namespace-uri() )">
+                <xsl:text>"</xsl:text><xsl:value-of select="local-name()"/><xsl:text>" : {</xsl:text>
+                <xsl:text>"</xsl:text>@type" : "<xsl:value-of select="namespace-uri()"/><xsl:text>"</xsl:text>
+                <xsl:text>}</xsl:text>
+
+                <xsl:variable name="sibs"
+                              select="following-sibling::*[not(self::atom:category) and
+                    not(self::atom:entry) and
+                    not(self::atom:feed) and
```
Comment (shintasmith) [inline src/main/xsl/xml2json-feeds.xsl]: Would we ever have a ```fn:archive``` with following-sibling ```atom:feed``` ?

File: src/main/xsl/xml2json-feeds.xsl (hunk 1 above)
Comment (ChandraAddala) [inline src/main/xsl/xml2json-feeds.xsl]: added tests

File: src/main/xsl/xml2json-feeds.xsl (hunk 2 above)
Comment (ChandraAddala) [inline src/main/xsl/xml2json-feeds.xsl]: @shintasmith @usnavi I addressed the comments and made some refactoring. I created an xslt function which seem to be working but if you guys(xslt masters) can take a closer look, I would appreciate it.

File: src/main/xsl/xml2json-feeds.xsl [hunk 3]
```diff
@@ -143,23 +143,24 @@
...
             <xsl:when test="count(./child::*) = 0">
                 <xsl:text>"</xsl:text><xsl:value-of select="local-name()"/>" : "<xsl:apply-templates select="."/><xsl:text>"</xsl:text>
-                <xsl:variable name="sibs" 
-                              select="following-sibling::*[not(self::atom:category) and
-                                                           not(self::atom:entry) and
-                                                           not(self::atom:feed) and
-                                                           not(self::atom:link)]"/>
-                <xsl:if test="count($sibs) &gt; 0">, </xsl:if>
+
+                <xsl:if test="cldfeeds:getSiblingCount(.) &gt; 0">, </xsl:if>
```
Comment (shintasmith) [inline src/main/xsl/xml2json-feeds.xsl]: The previous 2 calls to getSiblingCount() may not need to test the not(self::atom:feed), but this one may need it.

Comment (shintasmith) [inline src/main/xsl/xml2json-feeds.xsl]: Ummm... I take it back.... I can't think of a reason why atom:feed might have a sibling. He's an only child!

**Comment by shintasmith** [comment]
looks good to me, I will let @usnavi review and merge

**Comment by usnavi** [comment]
lgtm @ChandraAddala  - don't forget to update tiamat as well.

**Pull request merged**